File: 2021/May/Day 26/replace-words.py
# Source : https://leetcode.com/problems/replace-words/submissions/
class Solution:
    def replaceWords(self, dictionary: List[str], sentence: str) -> str:

        split = sentence.split(' ');

        for j in range(len(split)):
            for i in dictionary:
                if(split[j][:len(i)]==i):
                    split[j] =i;
        s="";
        for i in range(len(split)):
            s+=split[i];
            if(i!=len(split)-1):
                s+=" ";
        return s;

File: 2022/July/Day 1/replace-Words.cpp
//Source : https://leetcode.com/problems/replace-words/submissions/// Source : https://leetcode.com/problems/replace-words/submissions/
class Solution {
public:
    string replaceWords(vector<string>& dictionary, string sentence) {

        string out="";
        string temp="";

        sort(dictionary.begin(),dictionary.end());

        for(auto&i:sentence)
        {
            if(i!=' ')
            {
                temp+=i;
            }
            else
            {

                // Perfrom logic to check if the word is same as the root

                    for(auto&i:dictionary)
                    {
                        int length =size(i);
                        if(temp.substr(0,length)==i)
                        {
                            cout<<temp;
                            out+=i;
                            temp="";
                            break;
                        }


                    }
                   out+=temp+" ";
                temp ="";



            }




        }


        for(auto&i:dictionary)
                    {
                        int length =size(i);
                        if(temp.substr(0,length)==i)
                        {
                            cout<<temp;
                            out+=i;
                            temp="";
                            break;
                        }


                    }
                   out+=temp;
                temp ="";




        return out;


    }
};
